sourcegraph/cody
# Add prompt-like standard commands

Main Linear issue [SRCH-1052](https://linear.app/sourcegraph/issue/SRCH-1052/get-rid-of-out-of-the-box-commands-for-walmart-parent-task)
Closes [SRCH-1064](https://linear.app/sourcegraph/issue/SRCH-1064/automatically-create-ootb-prompts-for-plg-new-enterprise-prospects)
Closes [SRCH-1049](https://linear.app/sourcegraph/issue/SRCH-1049/kill-all-out-of-the-box-commands-put-behind-a-feature-flag-for-walmart)
Closes [SRCH-1050](https://linear.app/sourcegraph/issue/SRCH-1050/create-prompts-that-replace-todays-commands-and-automatically-load)

So this PR does a few things with prompts and commands (I recorded a loom with changes walkthrough [here](https://www.loom.com/share/fb8c8e18d34847eca248d9e0f729d59f))

- First of all, all changes in this PR are behind the feature flag `cody-unified-prompts` (I enabled it on s2 so the change will be available to you if you're connected to s2, if not, go to your instance, go to feature flags in site admin and create boolean feature flag `cody-unified-prompts`)

- If you enable this feature flag, you will see that standard commands become prompts and work as prompts now
   - Explain code just fills out the text box in the search 
   - Documentation code still works with selection but it also now works through chat 
   - Find smell and unit test generation as well as work through chat now

- Despite [SRCH-1049](https://linear.app/sourcegraph/issue/SRCH-1049/kill-all-out-of-the-box-commands-put-behind-a-feature-flag-for-walmart) You still can see these prompts in the context menu, originally we wanted to completely delete them for Walmart but since these commands now work as prompts we can keep them in sub-menus
   - But now out of box commands were renamed in Standard Prompts
   - The custom commands menu is hidden now if the feature flag is on 
   
- Keyboard shortcuts for commands-prompts like "Documentation" still work but it opens a new chat and fills out the textbox with a prompt about documentation 

## Tech note
Even though I called them "standard prompts," they still work on command rails from a tech point of view. However, having them as standard prompts requires a bit more work. It's a bit unclear how to properly get the current selection and current file (and in general context) to generate a proper text message. I think we will just use the same thing as we do with standard mentions but will try to find a less manual approach

## Test plan 
@chenkc805 @aramaraju detailed instruction if you want to check it locally https://www.loom.com/share/fb8c8e18d34847eca248d9e0f729d59f
 
- Open VSCode on this PR branch 
- Click the "Run and Debug" tabs in the left navigation panel 
- Click "Start debugging", this will open yet another VSCode instance with built local Cody extension 
- After this make sure that you're connected to S2, 
- All standard prompts like code documentation, or explain code,..etc should work as prompts
